### PR TITLE
Add service notes

### PR DESCRIPTION
**Pull request by CrystalPea**
Add service notes section to Service settings

EMPTY NOTES:
Screenshot 2021-01-15 at 16 26 57

ADDING SOME NOTES:
Screenshot 2021-01-15 at 16 26 42

NOTES PRESENT:
Screenshot 2021-01-15 at 16 26 31

LONG NOTES:
Screenshot 2021-01-15 at 16 29 23

### CODE REVIEW

File: app/main/views/service_settings.py [hunk 1]
```diff
@@ -1188,6 +1189,34 @@ def edit_data_retention(service_id, data_retention_id):
...
+
+        if form.notes.data == current_service.notes:
+            return redirect(url_for('.service_settings', service_id=service_id))
+
+        try:
+            current_service.update(
+                notes=form.notes.data
+            )
+        except HTTPError as e:
+            raise e
```
Comment (quis) [inline app/main/views/service_settings.py:1210]: I’m not sure you need the `try`/`except` here. Won’t the default error handling catch this and show a ‘technical difficulties’ page?

File: tests/app/main/views/test_service_settings.py [hunk 2]
```diff
@@ -5249,3 +5249,14 @@ def test_service_settings_links_to_edit_service_notes_page_for_platform_admins(
...
+
+def test_view_edit_service_notes(
+        platform_admin_client,
+        service_one,
+
+):
+    response = platform_admin_client.get(url_for('main.edit_service_notes', service_id=SERVICE_ONE_ID))
+    page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
+    assert page.select_one('h1').text == "Edit service notes"
+    assert page.find('label', class_="form-label").text.strip() == "Notes"
```
Comment (quis) [inline tests/app/main/views/test_service_settings.py:5262]: Would be good to add a test here to check that the textbox exists, and that it has the right `name` attribute

File: app/main/views/service_settings.py [hunk 3]
```diff
@@ -1188,6 +1189,34 @@ def edit_data_retention(service_id, data_retention_id):
     )
 
 
+@main.route("/services/<uuid:service_id>/notes", methods=['GET', 'POST'])
+@user_has_permissions('manage_service')
+def edit_service_notes(service_id):
+    form = EditServiceNotesForm()
+
+    if request.method == 'GET':
+        form.notes.data = current_service.notes
```
Comment (quis) [inline app/main/views/service_settings.py:1198]: You can achieve the same thing by passing `notes` as an argument to the form’s constructor:
```suggestion
    form = EditServiceNotesForm(notes=current_service.notes)
```
WTForms will make sure the `POST` data doesn’t get overridden with the existing value.

File: app/templates/components/table.html [hunk 4]
```diff
@@ -85,8 +86,8 @@
   </td>
 {%- endmacro %}
 
-{% macro text_field(text, status='', truncate=false) -%}
-  {% call field(status=status) %}
+{% macro text_field(text, status='', truncate=false, wrap=False) -%}
```
Comment (quis) [inline app/templates/components/table.html:89]: Nice attention to detail adding this option 👍🏻

Comment (CrystalPea) [reply app/templates/components/table.html:89]: Thanks 😊

File: tests/app/main/views/test_service_settings.py (hunk 2 above)
Comment (CrystalPea) [reply tests/app/main/views/test_service_settings.py:5262]: Good point, I will add that ☺️

File: app/main/views/service_settings.py (hunk 1 above)
Comment (CrystalPea) [reply app/main/views/service_settings.py:1207]: True, it should do.

**Review by quis — APPROVED** [review]

**Pull request merged**